dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+Field|Type|Required?|Description|
+|-|-|-|-|
+|merchant_data|MerchantDataObject|N|This element is required for some scenarios. It includes the details of the merchant to be displayed to the payer|
+|
+
+## PaymentActor Status Enum Extension
+The following values are added to the status enum of [PaymentActor](https://dip.diem.com/dip-1/#paymentactorobject) object:
+
+* `authorized` - This means the actor has performed all required tasks and funds are locked for a future capture. Note that the actual lock is done by the wallet (sender), so for the receiver this status means that all risk checks were successful and it is willing to accept the payment. 
+* `needs_init_data` - This means the actor requires basic data to initialize the payment. For the receiver, this status means that the payer data is required. For the sender, it means that the merchant data is required  
+
+For completeness purposes, the full list of values is: 
+* `none`
+* `needs_kyc_data`
+* `ready_for_settlement`
+* `abort`
+* `authorized`
+* `soft_match`
+* `needs_init_data`
+
+Allowed transition relevant for P2M:
+
+| Current Status | Next Status | Transition Comments |
+|-|-|-|
+| none | needs_init_data |Initial transition. Most common for P2M scenarios in which the actor requires additional data to initialize the payment|
+| needs_init_data | ready_for_settlement | All required tasks were performed and the payment is ready for settlement |
+| needs_init_data | authorized | All required tasks were performed and the funds are ready to be locked |
+| authorized | ready_for_settlement | Funds captured |
+| authorized | abort | Payment was voided |
+| needs_init_data | abort | Payment aborted before initial data was verified |
+| none | abort | Payment aborted before initialization |
+| none | ready_for_settlement | All required tasks were performed and the payment is ready for settlement. It may happen when no payer data is required |
+| none | authorized |  All required tasks were performed and the funds are ready to be locked. It may happen when no payer data is required |
+| ready_for_settlement | abort | One party aborted the payment even though the other party already notified it is ready to settle the payment. For example, it may happen if the sender does not approve merchant data provided by the receiver |
+|
+
+### AbortCodeEnum Extension
+
+Several P2M scenarios require adding the following values to the AbortCode Enum:
+* `void` - This abort code should be used for voiding a payment that was previously authorized. For example, the merchant voided the payment because the purchased products are out of stock. It is expected that the sender VASP will respond with void code as a sign of agreement
+* `risk_checks_failed` - This abort code should be used when the risk checks performed based on the payer information failed
+
+For completeness purposes, the full list of values of the abort code enum is:
+* `rejected`
+* `void`
+* `risk_checks_failed`
```

Comment:
if this is referring to dip-1, can we get a link?

Reply:
Sure.
Links will be added as soon as the content and structure of the DIP will be stable.